### DISCUSSION TITLE
Odometer Reading

### DISCUSSION THREAD

**Discussion by stu1811** [meatpiHQ/wican-fw#67] [Q&A]
Hello everyone. I just got my wican and trying it out with home assistant. I'm hoping to read the odometer. Does this look right?
```
msg.payload = { "bus": "0", "type": "tx", "frame": [{ "id": 2015, "dlc": 8, "rtr": false, "extd": false, "data": [2, 1, 166, 170, 170, 170, 170, 170] }] };
return msg;

From MQTT Explorer:
{
  "bus": "0",
  "type": "tx",
  "frame": [
    {
      "id": 2015,
      "dlc": 8,
      "rtr": false,
      "extd": false,
      "data": [
        2,
        1,
        166,
        170,
        170,
        170,
        170,
        170
      ]
    }
  ]
}
```

**Comment by meatpiHQ** [comment]
@stu1811 looks ok. Did you receive any response?

**Reply by stu1811** [reply]
I didn't get a response in node red. Also didn't see anything in mqtt explorer.

**Reply by meatpiHQ** [reply]
@stu1811 do you see any other frames on the CAN bus? Is the yellow LED blinking?

**Reply by stu1811** [reply]
I get a bunch of responses for 70 (ambient air) and 47 (fuel tank level). There are a few other ones I don't know.

{"bus":"0","type":"rx","ts":6755,"frame":[{"id":2024,"dlc":8,"rtr":false,"extd":false,"data":[3,65,70,58,0,0,0,0]}]}
{"bus":"0","type":"rx","ts":5120,"frame":[{"id":2024,"dlc":8,"rtr":false,"extd":false,"data":[3,65,47,214,0,0,0,0]}]}
{"bus":"0","type":"rx","ts":35157,"frame":[{"id":469533216,"dlc":8,"rtr":false,"extd":true,"data":[51,0,0,162,2,0,0,0]}]}
{"bus":"0","type":"rx","ts":5895,"frame":[{"id":2024,"dlc":8,"rtr":false,"extd":false,"data":[6,65,0,190,62,168,19,0]}]}

**Reply by meatpiHQ** [reply]
@stu1811 ok good.

It's possible that your car doesn't support this PID, try to use carscanner app, you can see the supported pids there.

**Comment by stu1811** [comment]
Look like odometer is not available but "Distance traveled since codes cleared" (PID 49) is available. I can make that work. Thanks!
